### PR TITLE
Improve filer ui

### LINKED ISSUE #571: Improve filer
- When there's no match, there shouldn't be a "NO MATCHES FOUND" message, it should be something like "Create file"
 - Back action (parent directory) doesn't clear the input
 - Make filer's path relative to either vim's cwd, or the user's home, if possible
 - Add keybinding to toggle between `filer` & `files`

### PR DESCRIPTION
**Pull request by romgrk**
Closes #571 

 - feat: make filer prompt relative to cwd or user home
 - fix: filer input cleared on back_action
 - feat: improve create-new-file entry
 - didn't add the keybinding to switch between files & filer because I'm too lazy

filer-dir
filer-new

### CODE REVIEW

File: autoload/clap/provider/filer.vim [hunk 1]
```diff
@@ -39,16 +40,22 @@ function! s:handle_response(result, error) abort
 endfunction
 
 function! s:set_prompt() abort
-  if strlen(s:current_dir) < s:winwidth * 3 / 4
-    call clap#spinner#set(s:current_dir)
+  let current_dir = s:current_dir
+  if stridx(current_dir, getcwd()) == 0
+    let current_dir = '.' . current_dir[len(getcwd()):]
```
Comment (liuchengxu) [inline autoload/clap/provider/filer.vim:45]:
```suggestion
  let cwd = getcwd()
  if stridx(current_dir, cwd) == 0
    let current_dir = '.' . current_dir[len(cwd):]
```

Comment (romgrk) [reply autoload/clap/provider/filer.vim]: :+1:

**Comment by liuchengxu** [comment]
We should also not send the on moved request on the CREATE FILE line, show something else in the preview window or close it.

```diff
diff --git a/autoload/clap/provider/filer.vim b/autoload/clap/provider/filer.vim
index 842f68f..5f74540 100644
--- a/autoload/clap/provider/filer.vim
+++ b/autoload/clap/provider/filer.vim
@@ -308,6 +308,10 @@ function! s:filer_handle_on_move_response(result, error) abort
 endfunction
 
 function! s:filer.on_move_async() abort
+  if g:clap.display.getcurline() =~# s:CREATE_FILE
+    call g:clap.preview.hide()
+    return
+  endif
   call clap#client#call_on_move('filer/on_move', function('s:filer_handle_on_move_response'), {'cwd': s:current_dir})
 endfunction
```

 The rest looks good to me.

**Comment by romgrk** [comment]
:+1:

**Pull request merged**